pandas/core/arrays/dates.py

```diff
@@ -0,0 +1,191 @@
+from pandas.core.dtypes.base import ExtensionDtype
+from pandas.core.arrays.datetimelike import DatelikeOps, DatetimeLikeArrayMixin
+from pandas.core.arrays.datetimes import sequence_to_dt64ns
+from pandas.core.dtypes.common import (
+    is_integer_dtype,
+    is_datetime64_dtype,
+    is_object_dtype,
+    pandas_dtype,
+)
+from pandas.core.dtypes.generic import ABCSeries, ABCIndexClass
+from pandas.core.dtypes.dtypes import DateDtype
+from pandas.core.construction import array
+from pandas._libs.tslibs import Timestamp
+from pandas._libs.tslibs.conversion import NS_DTYPE
+from pandas._libs import tslib, lib
+
+import numpy as np
+
+D_DATETIME_DTYPE = "datetime64[D]"
+INTEGER_BACKEND = "i8"
+VALID_TYPES = {INTEGER_BACKEND, "datetime64[ns]", D_DATETIME_DTYPE, "object"}
+
+
+def _to_date_values(values, copy=False):
+    data, _, _ = sequence_to_dt64ns(values, copy=copy)
+    return data.astype(D_DATETIME_DTYPE)
+
+
+class DateArray(DatetimeLikeArrayMixin, DatelikeOps):
+    """
+    Pandas ExtensionArray for date (year, month, day only) data.
+
+     .. warning::
+
+       DateArray is currently experimental, and its API may change
+       without warning. In particular, :attr:`DateArray.dtype` is
+       expected to change to always be an instance of an ``ExtensionDtype``
+       subclass.
+
+    Parameters
+    ----------
+    values : Series, Index, DateArray, ndarray
+        The date data.
+    freq : str or Offset, optional
+    dtype : pd.DateDtype
+
+    copy : bool, default False
+        Whether to copy the underlying array of values.
+
+    Attributes
+    ----------
+    None
+
+    Methods
+    -------
+    None
+    """
+
+    freq = "D"
+
+    def __init__(self, values, copy=False):
+        if isinstance(values, (ABCSeries, ABCIndexClass)):
+            values = values._values
+
+        if isinstance(values, type(self)):
+            values = values._data
+
+        if not isinstance(values, np.ndarray):
+            msg = (
+                f"Unexpected type '{type(values).__name__}'. 'values' must be "
+                "a DateArray ndarray, or Series or Index containing one of"
+                " those."
+            )
+            raise ValueError(msg)
+
+        if not self._is_compatible_dtype(values.dtype):
+            msg = (
+                f"The dtype of 'values' is incorrect. Must be one of {VALID_TYPES}."
+                f" Got {values.dtype} instead."
+            )
+            raise ValueError(msg)
+
+        if values.dtype == INTEGER_BACKEND:
```

Comment:
why aren't you simply keeping ordinals since epoch? its performant and *much* simpler

Reply:
I am keeping them, as I understand, the view just changes the outer representation, but not the backend. The same thing is done in the datetime array